content/docs/command-reference/remote/add.md

```diff
@@ -125,34 +125,40 @@ So, make sure you have the following permissions enabled:
 
 <details>
 
-### Click for S3 API compatible storage
+### Click for S3-compatible storage
 
-To communicate with a remote object storage that supports an S3 compatible API
+To communicate with a remote object storage that supports an S3-compatible API
 (e.g. [Minio](https://min.io/),
 [DigitalOcean Spaces](https://www.digitalocean.com/products/spaces/),
 [IBM Cloud Object Storage](https://www.ibm.com/cloud/object-storage) etc.),
-configure the remote's `endpointurl` explicitly:
-
-For example:
+configure the remote's `endpointurl`:
 
 ```dvc
 $ dvc remote add -d myremote s3://mybucket/path
 $ dvc remote modify myremote endpointurl \
                     https://object-storage.example.com
 ```
 
-> See `dvc remote modify` for a full list of S3 API parameters.
+It's also important to setup appropriate authentication with any of the related
+parameters available for regular S3 storage (see `dvc remote modify`).
+Otherwise, DVC may try to use default AWS credentials (if any), which will
+probably cause an error. Example:
+
+```dvc
+$ dvc remote modify myremote profile 'myprofile'
```

Comment:
Let's not push to specific methods in `remote add` (neither in `remote modify`).

I would also simplify the whole message. The is not that much specific about S3-compatible storage. Same happens when you have multiple S3 profiles. Regular credentials are being used and you have to specify auth options to make sure that the correct profile is set.

Reply:
I would replicate

> By default, DVC expects your AWS CLI is already configured. DVC will be using default AWS credentials file to access S3. To override some of these parameters, use the parameters described in dvc remote modify.

Add one sentence like - make sure that the it uses right credentials, especially if you have multiple S3 storages ...


But again, this problem happens with a regular S3 setup

Reply:
Agree, will simplify ⌛  But

> The is not that much specific about S3-compatible storage.

I think the special part is that you MUST setup auth with S3-compatible, as default AWS creds are pretty much guaranteed to fail. Maybe DVC shouldn't even try them if the `url` isn't .amazonaws.com ... But that's a separate question.

Reply:
OK I moved most info out of the S3-compatible sections. But I didn't replicate the same text as in the S3 section yet because, if we go that route, why have S3-compatible sections at all? If there truly isn't anything special about it, we should be able to merge the sections (just update the title a little). WDYT @shcheklein ?

Reply:
> If there truly isn't anything special about it

yes, the only special thing is the `endpointurl` + mention their names for SEO/search/visibility

everything else is not special

Reply:
> DVC shouldn't even try them if the url isn't .amazonaws.com

I would not do this. We don't know in advance the setup, people might use some proxy to access S3. And there is not way to say is it S3 credential or not - that's the whole point- they look the same as far as I understand, they come from the same file, same mechanism to use them, etc, etc

Reply:
> everything else is not special

@shcheklein I did also leave a warning about authentication at the bottom which I consider useful. But lmk if you think it should go and I'll follow up. Merging for now as I assume you looked again yet made no more comments + it's approved by @iesahin (ty BTW)

Reply:
My point about that note that it feels wrong to me. There is nothing wrong with this `DVC will try to use default AWS credentials, which may cause an error.`. And even more - that's what you want to do, you use the same regular mechanisms. I probably understand what you were trying to say here, but it's an edge case and should be written in a different way. Not it looks confusing (why would I even care about this?) or as a bug (that what I actually want).

Reply:
Hmmm. Based on a few support cases I thought that it's a bit more common than an edge case. Or do you mean you can use AWS credentials for Minio, DigitalOcean, etc? TBH I've never used those services.

OK @shcheklein, for now I copied the same note from the regular S3 section into here like you suggested in https://github.com/iterative/dvc.org/pull/2415#discussion_r619721921. And try to simplify again... See #2441

Reply:
You can use the same credentials _mechanisms_ (same config files, same env vars, etc). It's completely interchangeable with regular S3 in this way. The only thing that you need to provide is the endpointurl to override the default AWS's one.

For example, when you have multiple S3 accounts you will have to distinguish them somehow too (by specifying the `profile` name). Here is the same- if you use S3 _and_ minio you will have multiple profiles most likely.

Reply:
> You can use the same credentials mechanisms (same config files
when you have multiple S3 accounts you will have to distinguish them somehow

Right. But if you use AWS CLI (not necessarily for S3) \*and\* separately try to setup an S3-compatible remote by only changing the `endpoint` like we say, `push` fails because DVC tries to connect using `~/.aws/config`, which has Amazon-specific credentials. And the error message is about having wrong credentials, instead of a hint to configure the credentials (since none are set, really).

Perhaps it's an edge case (-compatible remotes aren't used often enough), not worth targeting in docs.

Reply:
p.s. there's an example support case, linked from the PR description.

Reply:
Same works for two S3 accounts - you specify URL s3://org1-some-bucket but it's using credentials from org2 (because it's set by default in the config). You will have the exact same error message, right?

Reply:
Yes.